app/models/prison.rb

```diff
@@ -47,7 +47,7 @@ def initialize(prison)
     def each
       number_of_requests = max_requests_count
 
-      (0..number_of_requests).each do |request_no|
+      (0..number_of_requests - 1).each do |request_no|
```

Comment:
Its a bug that has been around for ages, and which prevented me completing the story without adding lots more noise. I guess it would be nice to have a seperate PR for it.